scripts/migration/schema-0.2-GPII-4014/verify.js

```diff
@@ -90,76 +93,101 @@ gpii.migration.GPII4014.verifyAllDocs = function (options) {
+                    );
+                    if (!diffResult) {
+                        console.log("Error with the document _id \"" + aDoc._id + "\": new field values for client credential are incorrect - ", docFields);
+                        hasError = true;
+                    }
+                }
             }
-        }
-        if (hasError) {
-            numOfErrorDocs++;
-        }
-    });
-    console.log("The database has " + totalNumOfDocs + " documents. " + numOfVerified + " documents are qualified for verifications.");
-    if (numOfErrorDocs > 0) {
-        togo.reject("Fail: " + numOfErrorDocs + " documents have errors.");
-    } else {
-        togo.resolve("All passed.");
+            if (hasError) {
+                options.numOfErrorDocs++;
+            }
+        });
+        options.numOfVerified = Number(options.numOfVerified) + Number(allDocs.rows.length);
+        console.log("Verified " + options.numOfVerified + " of " + options.totalNumOfDocs + " documents.");
+        togo.resolve();
     }
     return togo;
 };
 
 /**
- * Create and execute the steps to verify documents.
+ * Verify recursively in small batches.
+ * @param {Object} options - Where to store information for verification as well as its progress:
+ * @param {Array} options.allDocsUrl - The url to query the database in small batches.
+ * @param {Array} options.novaClientCredentials - An array of NOVA client credential IDs.
+ * @param {Array} options.numOfVerified - The number of verified documents.
+ * @param {Array} options.numOfErrorDocs - The number of verified documents that have verification errors.
  */
-gpii.migration.GPII4014.verify = function () {
-    var options = gpii.migration.GPII4014.initOptions(process.argv);
-    var verifyPromise = gpii.migration.GPII4014.verifyAllDocs(options);
+gpii.migration.GPII4014.verifyRecursive = function (options) {
+    options.allDocsUrl = options.couchDbUrl + "/_all_docs?include_docs=true&descending=true&skip=" + options.numOfVerified + "&limit=" + options.maxDocsInBatchPerRequest;
+    var verifyPromise = gpii.migration.GPII4014.verifyDocsInBatch(options);
 
     verifyPromise.then(
-        function (result) {
-            console.log("Done - " + result);
-            process.exit(0);
+        function () {
+            gpii.migration.GPII4014.verifyRecursive(options);
         },
         function (error) {
-            console.log(error);
-            process.exit(1);
+            if (error.errorCode === "GPII-NO-MORE-DOCS") {
+                if (options.numOfErrorDocs > 0) {
+                    console.log("Fail: " + options.numOfErrorDocs + " documents have errors.");
+                } else {
+                    console.log("All passed.");
+                }
+                console.log("Done: " + error.message + " Verified " + options.totalNumOfDocs + " documents in total.");
+                process.exit(0);
```

Comment:
Push to top level

Reply:
This one still remains in old style. Also, now we have fixed the recursion algorithm, we can eliminate the strange "GPII-NO-MORE-DOCS" error payload and apply the standard resolve = success, reject = error semantics at top level.